Small changes to task argocd-task-sync-and-wait

jannfis opened the pull request:
Signed-off-by: Jann Fischer <jfischer@redhat.com>

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

This changes the container registry where Argo CD image is pulled from docker.io to quay.io (the project deprecated publishing on Docker Hub since a few releases, and will stop publishing new images to there soon). Also bumps the default version used by the task to the latest stable release.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

vdemeester commented:
/ok-to-test

vinamra28 reviewed (commented):
@jannfis, since the current task doesn't have any tests so just wanted to ask whether this will break existing users? Moreover, I can see that the default version is being bumped from `v1.0.2` to `v2.2.2` which seems to be a major change, was thinking whether this will be a good candidate of version bump of this task :thinking:

jannfis commented:
@vinamra28 Good question. I agree that this might justify a version bump in the task. 

I'm not familiar with Tekton and its ecosystem, but I stumbled over this today from a question that someone had and pointed me to this task. Argo CD v1.0.2 is pretty out of date (June 2019), so I would dare to say that the task as it is now might be breaking for everyone with a more or less recent Argo CD installation using the defaults provided.

As for backwards compatibility, the v2.2 CLI will work with the currently supported Argo CD releases (and most likely, with versions beyond support as well).

I'm happy to bump up the version number of the task as well. Is there anything special I need to know/take care of? Or just copycat the 0.1 directory to 0.2 and make my changes in the new file?

vinamra28 commented:
> @vinamra28 Good question. I agree that this might justify a version bump in the task.
> 
> I'm not familiar with Tekton and its ecosystem, but I stumbled over this today from a question that someone had and pointed me to this task. Argo CD v1.0.2 is pretty out of date (June 2019), so I would dare to say that the task as it is now might be breaking for everyone with a more or less recent Argo CD installation using the defaults provided.
> 
> As for backwards compatibility, the v2.2 CLI will work with the currently supported Argo CD releases (and most likely, with versions beyond support as well).
> 
> I'm happy to bump up the version number of the task as well. Is there anything special I need to know/take care of? Or just copycat the 0.1 directory to 0.2 and make my changes in the new file?

@jannfis, can we say that we are now deprecating `0.1` version of this task and bumping a new version? If yes, then you need to add an annotation `tekton.dev/deprecated: "true"` and before that, you need to copycat the 0.1 directory to 0.2 and make your changes in the respective file(s). Thanks for doing it :slightly_smiling_face:

jannfis commented:
Thanks @vinamra28 - deprecation of 0.1 task actually makes sense.

I pushed the changes to add deprecation annotation for the 0.1 task and added the bumped version and changed image registry as a new 0.2 task.

vinamra28 commented on task/argocd-task-sync-and-wait/0.2/README.md:
````diff
@@ -0,0 +1,77 @@
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/argocd-task-sync-and-wait/0.1/argocd-task-sync-and-wait.yaml
````
nit
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/argocd-task-sync-and-wait/0.2/argocd-task-sync-and-wait.yaml
```

jannfis replied:
Changed, thanks!

vinamra28 reviewed (commented):
overall PR looks good. Will it be possible for you to break this PR into 2 commits?
First commit to contain only 0.1 version changes and second commit to contain 0.2 version?

jannfis commented:
> Will it be possible for you to break this PR into 2 commits?

Split the change up into two separate commits. PTAL @vinamra28

vinamra28 commented:
/retest

jannfis commented:
/retest

jannfis commented:
I'm not sure why the CI test fails now @vinamra28 

Is that a problem with the PR or a flakyness in the CI? I'm wondering because in the previous (single-commit) incarnation, the tests have passed. Now, with nothing changed but splitting PR up into two commits, the CI seems to fail. :thinking:

vdemeester commented:
```
++ detect_changed_e2e_test
++ git --no-pager diff --name-only cbc2d9172cf23010c6249b10bf2b8e5adb0b1d76..9b5169da3f0958cde30afd5c2503e8eaa9c993c4
++ grep '^test/[^/]*'
+ [[ ! -z test/linux-ppc64le.sh ]]
+ TEST_RUN_ALL_TESTS=1
+ [[ -z 1 ]]
```

It re-runs everything because of this I think, and thus I would be on a CI flakyness. If you rebase on top of upstream main (aka https://github.com/tektoncd/catalog/commit/cbc2d9172cf23010c6249b10bf2b8e5adb0b1d76), it shouldn't detect this change and thus only execute the argocd task test 🤔

vinamra28 commented:
> ```
> ++ detect_changed_e2e_test
> ++ git --no-pager diff --name-only cbc2d9172cf23010c6249b10bf2b8e5adb0b1d76..9b5169da3f0958cde30afd5c2503e8eaa9c993c4
> ++ grep '^test/[^/]*'
> + [[ ! -z test/linux-ppc64le.sh ]]
> + TEST_RUN_ALL_TESTS=1
> + [[ -z 1 ]]
> ```
> 
> It re-runs everything because of this I think, and thus I would be on a CI flakyness. If you rebase on top of upstream main (aka [cbc2d91](https://github.com/tektoncd/catalog/commit/cbc2d9172cf23010c6249b10bf2b8e5adb0b1d76)), it shouldn't detect this change and thus only execute the argocd task test thinking

@jannfis, +1 to what @vdemeester said. Rebasing the PR on top of upstream main will help here.

vinamra28 commented:
/lgtm
/approve

vinamra28 commented:
/test .*

vinamra28 commented:
/test pull-catalog-catlin-lint
